R/spliceHelperFunctions.R

```diff
@@ -96,232 +96,180 @@ findSpliceOverlapsByDist <-function(query, subject, ignore.strand=FALSE, maxDist
+  ###############################PRE-PROCESSING#################################
   #for intron retention/exon skipping
-  querySplice.subjectFullQuerySplice.Rng <- rep(unlist(qSpRng, use.names=F),
-                                     rep(elementNROWS(subjectFullRng),
-                                         times=elementNROWS(qSpRng)))
-  subjectFullQuerySplice.Rng <- unlist(rep(subjectFullRng,
-                                                       times= elementNROWS(qSpRng)), use.names=F)
-  querySplice.subjectFullQuerySplice.IdMap <- rep(1:length(qSpRng),elementNROWS(qSpRng)*elementNROWS(subjectFullRng))
-  
-  subjectSplice.queryFullsubjectSplice.Rng <- rep(unlist(sSpRng, use.names=F),rep(elementNROWS(queryFullRng), times=elementNROWS(sSpRng)))
-  queryFullSubjectSplice.Rng <- unlist(rep(queryFullRng, times= elementNROWS(sSpRng)), use.names=F)
-  subjectSplice.queryFullSubjectSplice.IdMap <- rep(1:length(sSpRng), elementNROWS(sSpRng)*elementNROWS(queryFullRng))
-  
+  querySplice.subjectFullQuerySplice.Rng <- expandMatchedRanges(querySpliceRng, subjectFullRng)
+  subjectSplice.queryFullsubjectSplice.Rng <- expandMatchedRanges(subjectSpliceRng ,queryFullRng)
   #for exon skipping
-  
-  subjectStart.querySplice.Rng <- rep(unlist(subjectStartRng),elementNROWS(qSpRng))
-  subjectEnd.querySplice.Rng <- rep(unlist(subjectEndRng),elementNROWS(qSpRng))
-  subjectStartEnd.querySplice.IdMap <- rep(1:length(subjectStartRng),elementNROWS(qSpRng))
-
-  
-  queryStart.subjectSplice.Rng <- rep(unlist(queryStartRng),elementNROWS(sSpRng))
-  queryEnd.subjectSplice.Rng <- rep(unlist(queryEndRng),elementNROWS(sSpRng))
- queryStartEnd.subjectSplice.IdMap <- rep(1:length(queryStartRng),elementNROWS(sSpRng))
-  
-  
-  
-  
-  #internal start query
-  queryExonIntersect <- pintersect(queryStart.subjectFull.Rng,subjectList, resolve.empty='start.x')
-  startEndTable$internalFirstExon.query <- tapply(width(queryExonIntersect), queryStart.subjectFull.IdMap, sum)!= 0 & startEndTable$alternativeFirstExon
-  #internal start subject
-  subjectExonIntersect <- pintersect(subjectStart.queryFull.Rng,queryList, resolve.empty='start.x')
-  startEndTable$internalFirstExon.subject <- tapply(width(subjectExonIntersect), subjectStart.queryFull.IdMap, sum)!= 0 & startEndTable$alternativeFirstExon
-  #internal end query
-  queryExonIntersect=pintersect(queryEnd.subjectFull.Rng,subjectList, resolve.empty='start.x')
-  startEndTable$internalLastExon.query <- tapply(width(queryExonIntersect), queryEnd.subjectFull.IdMap, sum)!=0 & startEndTable$alternativeLastExon
-  #internal end subject
-  subjectExonIntersect=pintersect(subjectEnd.queryFull.Rng,queryList, resolve.empty='start.x')
-  startEndTable$internalLastExon.subject <- tapply(width(subjectExonIntersect), subjectEnd.queryFull.IdMap, sum)!=0 & startEndTable$alternativeLastExon
+  subjectStart.querySplice.Rng <- expandMatchedRanges(subjectStartRng, querySpliceRng)
```

Comment:
see comment above, shall we move this this the sub-functions?

Reply:
The expandMatchedRanges() are now called from sub-functions.